app/src/main/java/com/example/cocktails/presentation/fragments/details/CocktailDetailsFragment.kt

```diff
@@ -0,0 +1,162 @@
+                        .setTitle("Error")
+                        .setPositiveButton("OK") { _, _ ->
+                        }
+
+                    val dialog: AlertDialog = builder.create()
+                    dialog.show()
+                }
+            }
+        }
+
+        cocktailDetailsViewModel.fetchAllCocktailDetails(argId)
+
+    }
+
+    private fun createList(cocktailDetails: List<CocktailDetails>) {
+        cocktailList = cocktailDetails.map { cocktailDetailsItem ->
+            Cocktail(
+                id = cocktailDetailsItem.idDrink,
+                name = cocktailDetailsItem.strDrink,
+                image = cocktailDetailsItem.strDrinkThumb,
+                alcoholic = cocktailDetailsItem.strAlcoholic,
+                favorite = argFavorite,
+                email = argEmail
+            )
+        }
+    }
+
+    private fun initUI() {
+        Glide.with(binding.cocktailImage.context)
+            .load(cocktailDetailsMap?.get("strDrinkThumb"))
+            .into(binding.cocktailImage)
+
+        binding.cocktailName.text = cocktailDetailsMap?.get("strDrink")
+        binding.labelAlcoholic.text = cocktailDetailsMap?.get("strAlcoholic")
+        binding.labelCategory.text = cocktailDetailsMap?.get("strCategory")
+        binding.labelGlass.text = cocktailDetailsMap?.get("strGlass")
+        binding.ingredientsItems.text = stringBuilderIngredient
+        binding.instructionsItems.text = cocktailDetailsMap?.get("strInstructions")
+        setImage()
+
+        binding.favoriteButton.setOnClickListener {
+            if (argFavorite) {
+                cocktailDetailsViewModel.deleteCocktail(cocktailList.first())
+            } else {
+                cocktailDetailsViewModel.insertCocktail(cocktailList.first())
+            }
+            argFavorite = !argFavorite
+            setImage()
+        }
+    }
+
+    private fun setImage() {
+        val imageResourceId = if (argFavorite) {
+            R.drawable.imageview_checked
+        } else {
+            R.drawable.imageview_unchecked
+        }
+        Glide.with(binding.favoriteButton.context).load(imageResourceId)
+            .into(binding.favoriteButton)
+    }
+
+    private fun getNonNullFields(cocktailDetailsList: List<CocktailDetails>): Map<String, String>? {
+        return cocktailDetailsList.firstOrNull()?.let { cocktailDetails ->
+            val fields = CocktailDetails::class.java.declaredFields
+            val result = mutableMapOf<String, String>()
+
+            for (field in fields) {
+                field.isAccessible = true
+                val value = field.get(cocktailDetails)
+                if (value != null) {
+                    result[field.name] = value.toString()
+                }
+            }
+
+            result.ifEmpty { null }
+        }
+    }
+
+    private fun setDetails() {
+        for (i in 1..15) {
+            val ingredientKey = "strIngredient$i"
```

Comment:
Also extract to a constant as previous comment. Also, implement the missing logic for showing MEASURES.